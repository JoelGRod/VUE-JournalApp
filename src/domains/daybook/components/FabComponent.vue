<template>
  <button class="btn btn-primary fab-button">
      <i :class="getIcon"></i>
  </button>
</template>

<script>
export default {
  name: 'fab-icon',
  props: {
    icon: {
      type: String,
      default: 'fa-plus'
    }
  },
  computed: {
    getIcon() {
      return `fa fa-2x ${ this.icon }`
    }
  }

}
</script>

<style lang="scss" scoped>

.fab-button {
    position: fixed;
    bottom: 2rem;
    right: 2rem;
    width: 4rem;
    height: 4rem;
    border-radius: 100%;
}

</style>