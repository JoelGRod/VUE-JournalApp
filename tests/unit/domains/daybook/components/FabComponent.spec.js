import { shallowMount } from "@vue/test-utils";
import FabComponent from "@/domains/daybook/components/FabComponent";

describe('Testing FabComponent.vue', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallowMount( FabComponent );
    });

    test('should match with the snapshot', () => {
        expect( wrapper.html() ).toMatchSnapshot();
    });

    test('should show the default icon (fa-plus)', () => {
        
    });

    test('should show the fa-circle icon', () => {
        
    });

    test('should emit on:click event when click', () => {
        
    });
})