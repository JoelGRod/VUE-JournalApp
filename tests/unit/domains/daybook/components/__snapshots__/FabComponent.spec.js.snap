// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FabComponent.vue should match with the snapshot 1`] = `<button class="btn btn-primary fab-button"><i class="fa fa-2x fa-plus"></i></button>`;
